File: FrontEnd (C#)/SoftProgBusiness/GestPrestamos/BOI/PrestamoBOImpl.cs
using SoftProgBusiness.GestPrestamos.BO;
using SoftProgModel.GestPrestamos;
using SoftProgPersistance.GestPrestamos.DAO;
using SoftProgPersistance.GestPrestamos.Impl;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SoftProgBusiness.GestPrestamos.BOI
{
    public class PrestamoBOImpl : IPrestamoBO
    {
        private PrestamoDAO daoPrestamo;
        public PrestamoBOImpl()
        {
            daoPrestamo = new PrestamoImpl();
        }
        public int eliminar(int idObjeto)
        {
            return daoPrestamo.eliminar(idObjeto);
        }

        public int finalizar_prestamo(int idPrestamo)
        {
            return daoPrestamo.finalizar_prestamo(idPrestamo);
        }

        public int insertar(Prestamo objeto)
        {
            return daoPrestamo.insertar(objeto);
        }

        public BindingList<Prestamo> listarTodos()
        {
            return daoPrestamo.listarTodos();
        }

        public int modificar(Prestamo objeto)
        {
            return daoPrestamo.modificar(objeto);   
        }

        public Prestamo obtenerPorId(int idObjeto)
        {
            return daoPrestamo.obtenerPorId (idObjeto); 
        }

        public void validar(Prestamo objeto)
        {
            throw new NotImplementedException();
        }
    }
}

File: BackEnd (Java)/BibliotecaBusiness/src/main/java/biblioteca/gestionPrestamo/boImpl/PrestamoBOImpl.java

package biblioteca.gestionPrestamo.boImpl;

import biblioteca.gestionPrestamo.bo.PrestamoBO;
import biblioteca.gestionPrestamo.dao.PrestamoDAO;
import biblioteca.gestionPrestamo.mysql.PrestamoImpl;
import java.util.ArrayList;
import pe.edu.pucp.utilsarmy.gestion_de_prestamos.model.Prestamo;

public class PrestamoBOImpl implements PrestamoBO{
    private final PrestamoDAO daoPrestamo;

    public PrestamoBOImpl() {
        daoPrestamo = new PrestamoImpl();
    }

    @Override
    public int insertar(Prestamo objeto) throws Exception {
        return daoPrestamo.insertar(objeto);
    }

    @Override
    public int modificar(Prestamo objeto) throws Exception {
        return daoPrestamo.modificar(objeto);
    }

    @Override
    public int eliminar(int idObjeto) throws Exception {
        return daoPrestamo.eliminar(idObjeto);
    }

    @Override
    public Prestamo obtenerPorId(int idObjeto) throws Exception {
        return daoPrestamo.obtenerPorId(idObjeto);
    }

    @Override
    public ArrayList<Prestamo> listarTodos() throws Exception {
        return daoPrestamo.listarTodos();
    }

    @Override
    public void validar(Prestamo objeto) throws Exception {
        throw new UnsupportedOperationException("Not supported yet."); // Generated from nbfs://nbhost/SystemFileSystem/Templates/Classes/Code/GeneratedMethodBody
    }

    @Override
    public ArrayList<Prestamo> listarPorUsuario(int idUsuario) {
        return daoPrestamo.listarPorUsuario(idUsuario);
    }
    
}
